documentation/api.md

```diff
@@ -152,10 +152,115 @@ See example response.
 
 # Assignments
 
+## Assignment Info
+>><h4> Example Response </h4>
+> ```
+curl "https://ok.cs61a.org/api/v3/assignment/cal/cs61a/sp16/lab01"
+{
+    "code": 200,
+    "data": {
+        "course": {
+            "active": true,
+            "display_name": "CS 61A",
+            "id": 1,
+            "offering": "cal/cs61a/sp16"
+        },
+        "display_name": "Lab 01",
+        "due_date": "2016-09-07T06:59:59",
+        "files": {
+            "fizzbuzz.py": "if ..."
+        },
+        "max_group_size": 2,
+        "name": "cal/cs61a/sp16/lab01",
+        "url": null
+    },
+    "message": "success"
+}
+```
+
+Get detailed assignment info.
+
+#### Permissions
+Requires an access token. The assignment is presented to all staff members of the course or admins but is only shown to other users if the staff marked the assignment as visible .
+
+#### HTTP Request
+`GET https://ok.cs61a.org/api/v3/assignment/<assignment_name:endpoint>``
+
+#### Query Parameters
+Parameter | Default | Description
+---------- | ------- | -------
+access_token | None | (Required) Access Token of staff member
+
+#### Response
+Name | Type | Description
+---------- | -------
+due_date | DateTime | UTC Time of deadline in iso8601 format.
+files | Dictionary | Template file names and contents
+name | String | Endpoint
+
+See example output for other fields.
+
+## Group Info
+>><h4> Example Response </h4>
+> ```
+curl "https://ok.cs61a.org/api/v3/assignment/cal/cs61a/sp16/lab01/group/email@example.com"
+{
+    "code": 200,
+    "data": {
+        "assignment": {
```

Comment:
We shouldn't send assignment info, because this will send the file templates as well. The client should make a separate request for the assignment info and cache that. In general, I'd like to keep the endpoints lightweight to avoid the get_course_info fiasco.